app.go

```diff
@@ -21,7 +27,28 @@
 	a.ctx = ctx
 }
 
-// Greet returns a greeting for the given name
-func (a *App) Greet(name string) string {
-	return fmt.Sprintf("Hello %s, It's show time!", name)
+func (a *App) Greet(url string) string {
+
+	_, err := a.GithubService.ParsePullRequestURL(url)
+
+	if err != nil {
+
+		return fmt.Errorf("invalid GitHub pull request url. error: %s", err).Error()
+	}
+
+	pullRequest, err := a.GithubService.GetPullRequest(context.TODO(), url)
```

Comment:
This is a side effect of the wails architecture.  I will look at how to best manage the context.

Reply:
Out of curiosity, given the greater scope, how would you best manage context?

I would argue context is quite disconnected from the starting point (the user's interaction with a DOM element) which is in javascript land.  

In that case, the async function call in javascript has its own timeout.  This realistically cannot be communicated to the back end without very tight coupling.  As a result, these two timeouts being out of sync becomes somewhat challenging to manage and debug. 

Now, that doesn't get our back-end function off the hook from managing its own outbound connection timeouts, and somehow aligning this with the front end.  For instance, if the front end times out, and the back end returns, what happens to its return values?

At this point, we have to ask ourselves, what are we trying to accomplish.

Remember, this is in the scope of the bindings between go and typescript in the Wails environment.

Reply:
A message queue is quite compelling.  What facilities do we have at our dispsal?

Reply:
We're talking about a message queue between javascript and golang via their bindings in wails. 

This would be some kind of local in memory queue data structure, is what I'm picturing.  But who brokers it, and what manages the bridge between the two runtimes?

How would a queue system in this context work?

Reply:
Is it fair, then, to say that we should think about the interface between typescript and golang in the context of wails, is much like we'd think of a stateless rest api and a consumer?

Reply:
Well I'll be an s.o.b.  That'll do just fine.

Reply:
@frankie-mur ☝️ we'll end up accidentally spitting out completion instructions on accident also :)